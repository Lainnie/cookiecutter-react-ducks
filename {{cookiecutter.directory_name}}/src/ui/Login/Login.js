import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { Field, reduxForm } from 'redux-form';

import { authOperations, authSelectors } from '../../core/auth';
import { TextField } from '../Lib';

export class LoginPage extends React.Component {
  onSubmit = (values = {}) => {
    this.props.login(values.username, values.password);
  };

  render() {
    return <LoginView {...this.props} onSubmit={this.onSubmit} />;
  }
}

const LoginView = ({ onSubmit, submitError }) => (
  <Container className="background_login">
    <Logo src="images/logo_login.png" />
    <LoginForm onSubmit={onSubmit} submitError={submitError} />
  </Container>
);

const FormView = ({ handleSubmit, submitError }) => (
  <Form className="card" onSubmit={handleSubmit}>
    <div>
      <label htmlFor="username">Username</label>
      <TextField name="username" component="input" type="text" />
    </div>
    <div>
      <label htmlFor="password">Password</label>
      <Field className="" name="password" component="input" type="password" />
    </div>
    {submitError && (
      <strong className="flow-text red-text text-darken-2">
        Wrong credentials
      </strong>
    )}
    <button className="waves-effect waves-light btn" type="submit">
      Submit
    </button>
  </Form>
);

const LoginForm = reduxForm({
  form: 'login',
})(FormView);

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 100%;
`;

const Logo = styled.img`
  display: block;
  width: 210px;
  height: auto;
  margin: 40px auto;
`;

const Form = styled.form`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 400px;
  padding: 40px 0;
`;

const mapStateToProps = state => ({
  submitError: authSelectors.getSubmitError(state),
});
const mapDispatchToProps = {
  login: authOperations.login,
};

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);
