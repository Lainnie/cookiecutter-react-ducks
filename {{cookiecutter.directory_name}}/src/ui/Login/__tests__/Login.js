import React from 'react';
import shallow from 'enzyme/shallow';

import { LoginPage as Login } from '../Login';

describe('Login', () => {
  it('renders correctly', () => {
    const tree = shallow(<Login />).dive();

    expect(tree).toMatchSnapshot();
  });

  describe('when submitting form', () => {
    const login = jest.fn();

    it('calls authOperations.login with usermame and password', () => {
      const tree = shallow(<Login login={login} />);
      const instance = tree.instance();

      instance.onSubmit({ username: 'username', password: 'password' });

      expect(login).toHaveBeenCalledWith('username', 'password');
    });
  });
});
