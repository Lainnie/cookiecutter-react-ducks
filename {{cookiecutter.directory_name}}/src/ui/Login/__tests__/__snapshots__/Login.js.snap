// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<styled.div
  className="background_login"
>
  <styled.img
    src="images/logo_login.png"
  />
  <ReduxForm
    onSubmit={[Function]}
  />
</styled.div>
`;
